File: go/leetcode/two_sum.go
package leetcode

func twoSum(nums []int, target int) [2]int {
	hash := make(map[int]int, len(nums))

	for i, v := range nums {
		hashItem, ok := hash[target-v]
		if ok {
			return [2]int{hashItem, i}
		}

		hash[nums[i]] = i
	}

	return [2]int{}
}

File: python/src/leetcode/two_sum.py
from types import NoneType


def two_sum(nums: list[int], target: int) -> tuple[int, int]:
    hash: dict[int, int] = {}

    for i, v in enumerate(nums):
        item = hash[target - v] if target - v in hash else None
        if type(item) != NoneType:
            return item, i

        hash[v] = i

    return tuple()
